examples/ordering/src/network.rs

```diff
@@ -0,0 +1,80 @@
+use aleph_bft::Recipient;
+use aleph_bft_mock::{Data, Hasher64, PartialMultisignature, Signature};
+use codec::{Decode, Encode};
+use log::error;
+use std::{io::Write, net::SocketAddr};
+use tokio::{
+    io::{self, AsyncReadExt},
+    net::TcpListener,
+};
+
+pub type NetworkData = aleph_bft::NetworkData<Hasher64, Data, Signature, PartialMultisignature>;
+
+pub struct Network {
+    my_id: usize,
+    addresses: Vec<SocketAddr>,
+    listener: TcpListener,
+}
+
+impl Network {
+    pub async fn new(my_id: usize, ports: &[usize]) -> Result<Self, Box<dyn std::error::Error>> {
+        assert!(my_id < ports.len());
+        let mut addresses = vec![];
+        for port in ports {
+            addresses.push(format!("127.0.0.1:{}", port).parse::<SocketAddr>()?);
+        }
+        let listener = TcpListener::bind(format!("127.0.0.1:{}", ports[my_id])).await?;
+        Ok(Network {
+            my_id,
+            addresses,
+            listener,
+        })
+    }
+
+    fn send(&self, data: NetworkData, recipient: &Recipient) -> io::Result<()> {
+        match recipient {
+            Recipient::Everyone => {
+                for r in 0..self.addresses.len() {
+                    self.send_to_peer(data.clone(), r)?;
+                }
+            }
+            Recipient::Node(r) => self.send_to_peer(data, r.0)?,
+        }
+        Ok(())
+    }
+
+    fn send_to_peer(&self, data: NetworkData, recipient: usize) -> io::Result<()> {
+        assert!(recipient < self.addresses.len());
+        if recipient != self.my_id {
```

Comment:
I don't think we should hide not sending messages to ourself from AlephBFT. I would remove this if and this shouldn't break anything, right?

Reply:
fixed